src/planwise/client/scenarios/edit.cljs

```diff
@@ -26,31 +26,29 @@
                :accept-fn  #(dispatch [:scenarios/accept-rename-dialog])
                :cancel-fn  #(dispatch [:scenarios/cancel-rename-dialog])}))))
 
-(defn input
-  [{:keys [value onchange-path]}]
-  [common2/text-field {:type "number"
-                       :on-change  #(dispatch [:scenarios/save-key onchange-path (-> % .-target .-value int)])
-                       :value value}])
-
 (defn changeset-dialog-content
-  [{:keys [investment capacity]}]
-  [:div
-   [:h2 "Investment"]
-   [input {:value (or investment "")
-           :onchange-path [:changeset-dialog :investment]}]
-   [:h2 "Capacity"]
-   [input {:value (or capacity "")
-           :onchange-path [:changeset-dialog :capacity]}]])
+  [{:keys [investment capacity]} scenario-investment project-budget]
```

Comment:
It's better to receive a `remaining-budget` or `available-budget` here. That's one less parameter and clearer what the purpose of the additional data is.